otelcol/configprovider.go

```diff
@@ -14,6 +14,16 @@ import (
 	"go.opentelemetry.io/collector/confmap/provider/httpprovider"
 	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
 	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+// PreventEnvVarExpansionFeatureGate is the feature gate that controls whether the collector
+// supports configuring the OpenTelemetry SDK via configuration
+var PreventEnvVarExpansionFeatureGate = featuregate.GlobalRegistry().MustRegister(
+	"confmap.preventEnvVarExpansion",
```

Comment:
```suggestion
	"confmap.NoBracelessEnvVars",
```
I think we should refine the name a bit, since we will still allow env var expansion even after the deprecation process. We could also say `NoNakedEnvVars`, but I think "braceless" is a bit clearer to someone who isn't familiar with our terminology.

Reply:
Should we have a single feature gate for all changes related to the environment variable revamp? Or should we have one for each change? I am inclined to say we should have one (easier to figure out how to revert back to the old behavior without having to understand its intricacies)

Reply:
What is another change? I thought `$ENV` was the only thing we were deprecating

Reply:
Making types stricter and unified between `${ENV}` and `${env:ENV}` is the other part